daffodil-runtime2/src/test/c/examples/variablelen/generated_code.h

```diff
@@ -0,0 +1,22 @@
+#ifndef GENERATED_CODE_H
+#define GENERATED_CODE_H
+
+// clang-format off
+#include <stdbool.h>  // for bool
+#include <stddef.h>   // for size_t
+#include <stdint.h>   // for uint8_t, int16_t, int32_t, int64_t, uint32_t, int8_t, uint16_t, uint64_t
+#include "infoset.h"  // for InfosetBase, HexBinary
+// clang-format on
+
+// Define infoset structures
+
+typedef struct expressionElement_
+{
+    InfosetBase _base;
+    uint32_t    before;
+    uint32_t    variablelen_size;
+    uint32_t    variablelen[16];
```

Comment:
I've implemented a similar idea in HexBinary, which has a `bool dynamic` field recording whether its `array` pointer points to static memory or separately allocated memory on the heap.  I didn't do that in this PR since we're unlikely to run the code generator on DFDL schemas using large or unbounded arrays at this time and we can do another PR if that does occur.